app/Services/Widgets/Video.php

```diff
@@ -29,6 +29,29 @@ public static function getFromWidgetDom($widgetDom) {
     return $widgetData;
   }
 
+  public static function getVideoFromHeader($provider, $videoId) {
+    $widgetData        = new stdClass();
+    $widgetData->type  = 'embed';
+
+    switch ( $provider ) {
+      case 'youtube':
+        $widgetData->embed = "https://www.youtube.com/watch?v=".$videoId;
+        break;
+      // case 'jwplayer':
+        // $widgetData->embed = "https://www.jwplayer.com/".$videoId;
+        // break;
+      case 'vimeo':
+        $widgetData->embed = "https://vimeo.com/".$videoId;
+        break;
+      // case 'facebook':
+        // $widgetData->embed = "https://www.facebook.com/watch/".$videoId;
+        // break;
+    }
```

Comment:
Good point, I can help with this. In Catfish, the video ID for a Facebook video is actually the whole URL. This is confusing/wrong on their part.
So I think you can just do this for FB video: `$widgetData->embed = $videoId;`